bip-XXXX.mediawiki

```diff
@@ -0,0 +1,65 @@
+<pre>
+  BIP: XXXX
+  Layer: Consensus (hard fork)
+  Title: Limit number of transactions which contain an OP_RETURN opcode to one per block
+  Author: Christopher Gilliard <christopher.gilliard@gmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-XXXX
+  Status: Draft
+  Type: Standards Track
+  Created: 2021-04-16
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This BIP proposes limiting the number of transactions which contain an OP_RETURN opcode to one per block.
+
+==Motivation==
+
+For background, the OP_RETURN opcode [<a href="#references">1</a>], is used in Bitcoin to store arbitrary additional data into a block which is then permanently stored as part of the blockchain. Today there is a limitation of one OP_RETURN per transaction and each OP_RETURN is limited to 80 bytes of data. With very limited block space, it is essential to optimize the data stored in a block. Since a Merkle Tree [<a href="#references">2</a>] or other similar data structures can be used to timestamp multiple documents by storing a single 32 byte checksum of the top of a Merkle Tree in a block, there is no reason to allow more than one OP_RETURN per block. If only a single OP_RETURN is allowed, per block, users will be required to coordinate in order to store individual checksums into this data structure, but this can be achieved by developing a layer two time-stamping protocol. This layer two time-stamping protocol will allow users to store thousands or even millions of checksums into a single block and at the same time limit the total amount of data in the block to 80 bytes. A future BIP proposing such a layer two protocol will be forthcoming. Use cases other than time-stamping will be harder to achieve with this proposal, and that is part of the goal of this BIP since those other use cases should be discouraged by the rules of the protocol.
```

Comment:
Note: 80 bytes is a particular node-fiat policy, not part of the Bitcoin consensus system.